### PR TITLE
Support Scan Breakglass

### PR DESCRIPTION
**Pull request by scbedd**
Getting a simple feature in ahead of the need.

I can definitely see us doing something more specific, like an internal suppression mechanism or the like, but I want to be able to unblock folks on their machines if they absolutely need it.

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/CommandOptions/OptionsGenerator.cs [hunk 1]
```diff
@@ -55,6 +55,11 @@ public static RootCommand GenerateCommandLineOptions(Func<DefaultOptions, Task>
                 getDefaultValue: () => false);
             universalOption.AddAlias("-u");
 
+            var breakGlassOption = new Option<bool>(
+                name: "--breakGlass",
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/CommandOptions/OptionsGenerator.cs:59]:
```suggestion
                name: "--break-glass",
```

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/CommandOptions/OptionsGenerator.cs [hunk 2]
```diff
@@ -55,6 +55,11 @@ public static RootCommand GenerateCommandLineOptions(Func<DefaultOptions, Task>
...
+                description: "Flag; Ignore secret push protection results when pushing.",
+                getDefaultValue: () => false);
+            breakGlassOption.AddAlias("-b");
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/CommandOptions/OptionsGenerator.cs:62]: For stuff like this I would say let's not add a shorthand alias.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs [hunk 3]
```diff
@@ -145,8 +146,11 @@ public bool CheckForSecrets(GitAssetsConfiguration assetsConfiguration, string[]
             {
                 if (CheckForSecrets(config, pendingChanges))
                 {
-                    Environment.ExitCode = -1;
-                    return;
+                    if (!ignoreSecretProtection)
+                    {
+                        Environment.ExitCode = -1;
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs:151]: Is this a not implemented type thing right now?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/CommandOptions/OptionsGenerator.cs (hunk 2 above)
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/CommandOptions/OptionsGenerator.cs:62]:
```suggestion
```

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs (hunk 3 above)
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs:151]: I don't get what you're implying here.

**Review by benbp — APPROVED** [review]

**Pull request merged**